src/groovy/com/recomdata/extensions/ExtensionsRegistry.groovy

```diff
@@ -1,13 +1,74 @@
 package com.recomdata.extensions
 
+import groovy.transform.CompileStatic
+
 /**
  * Date: 14-Jan-16
  * Time: 18:08
  */
 class ExtensionsRegistry {
     def analysisTabExtensions = []
+    List<Map<String, Object>> tabs = new LinkedList<>()
+
+    ExtensionsRegistry() {
+    }
 
     void registerAnalysisTabExtension(Map<String, Object> config = [:], String extensionId, String resourcesUrl, String bootstrapFunction) {
         analysisTabExtensions.add([extensionId: extensionId, resourcesUrl: resourcesUrl, bootstrapFunction: bootstrapFunction, config: config])
     }
+
+    void registerTab(Map<String, Object> config = [:], String id, String title, String controller) {
+        def tab = [id: id, title: title, controller: controller, insertBefore: config.insertBefore]
+        insertBeforeFirstReference(tabs, tab)
+    }
+
+    @CompileStatic
+    private static boolean contains(Object obj, Object item) {
+        obj instanceof Collection ? ((Collection) obj).contains(item) : obj == item
+    }
```

Comment:
Yes, thank you! I remember something like this was in Groovy, but didn't find it. I will update code using `isCase` operator.